community/kernel/src/test/java/org/neo4j/test/Race.java

```diff
@@ -118,5 +121,14 @@ public void run()
                 throw e;
             }
         }
+
+        private void randomlyDelaySlightly()
+        {
+            int target = ThreadLocalRandom.current().nextInt( 1_000_000_000 );
+            for ( int i = 0; i < target; i++ )
+            {
+                i = i;
+            }
+        }
```

Comment:
Thought about it, but I've seen it working before. It's probably sensitive to HIT and environment in general as you say. Would be great if you could replace this with what you had in mind.